File: Android/app/src/main/java/com/checksheet/android/data/model/SectionEquipmentMap.kt
package com.checksheet.android.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class SectionEquipmentMap(
    val id: Int,
    @SerialName("section_id") val sectionId: Int,
    @SerialName("equipment_id") val equipmentId: Int,
    val technology: String,
    @SerialName("is_active") val isActive: Boolean,
    @SerialName("section_name") val sectionName: String? = null,
    @SerialName("equipment_name") val equipmentName: String? = null
)

File: Backend/app/models/section_equipment_map.py
from sqlalchemy import Boolean, Column, ForeignKey, Integer, String
from sqlalchemy.orm import relationship

from app.database.base import Base


class SectionEquipmentMap(Base):
    __tablename__ = "section_equipment_map"

    id = Column(Integer, primary_key=True, index=True)

    section_id = Column(Integer, ForeignKey("sections.id"), nullable=False)

    equipment_id = Column(Integer, ForeignKey("equipment.id"), nullable=False)

    technology = Column(String(20), nullable=False)

    is_active = Column(Boolean, default=True)

    section = relationship("Section")
    equipment = relationship("Equipment")

    @property
    def section_name(self):
        return self.section.name if self.section else None

    @property
    def equipment_name(self):
        return self.equipment.equipment_name if self.equipment else None
